From: Guillaume Delacour <gui@iroqwa.org>
Subject: rsync --rsh problem because arguments are now performed via perl execvp
Origin: upstream, https://github.com/DrHyde/rsnapshot/pull/1
Forwarded: not-needed
Bug-Debian: http://bugs.debian.org/717451

--- a/rsnapshot-program.pl
+++ b/rsnapshot-program.pl
@@ -3452,7 +3452,7 @@
 		
 		# if we have any args for SSH, add them
 		if ( defined($ssh_args) ) {
-			push( @rsync_long_args_stack, "--rsh=\"$config_vars{'cmd_ssh'} $ssh_args\"" );
+			push( @rsync_long_args_stack, "--rsh=$config_vars{'cmd_ssh'} $ssh_args" );
 			
 		# no arguments is the default
 		} else {
