From: Guillaume Delacour <gui@iroqwa.org>
Subject: Document rsnapshot configtest argument
Bug-Debian: https://bugs.debian.org/720509
Forwarded: yes, https://github.com/rsnapshot/rsnapshot/pull/80

Index: rsnapshot/rsnapshot-program.pl
===================================================================
--- rsnapshot.orig/rsnapshot-program.pl
+++ rsnapshot/rsnapshot-program.pl
@@ -6873,6 +6873,14 @@ run rsnapshot again, using example.com/e
 
 =back
 
+B<rsnapshot configtest>
+
+=over 4
+
+Do a quick sanity check to make sure everything is ready to go.
+
+=back
+
 =head1 EXIT VALUES
 
 =over 4
