From: Julius Seemayer <debian@yeeer.net>
Subject: Silence lvremove command in linux_lvm_cmd_lvremove
Bug-Debian: https://bugs.debian.org/745963
Forwarded: no

Index: rsnapshot/rsnapshot-program.pl
===================================================================
--- rsnapshot.orig/rsnapshot-program.pl
+++ rsnapshot/rsnapshot-program.pl
@@ -3727,9 +3727,7 @@ sub rsync_backup_point {
         
         print_cmd(@cmd_stack);
         if (0 == $test) {
-            # silence gratuitous lvremove output
-            #$result = system(@cmd_stack);
-            $result = system(join " ", @cmd_stack, ">/dev/null");
+            $result = system(@cmd_stack);
             
             if ($result != 0) {
                 bail("Unmount LVM snapshot failed: $result");
@@ -3744,7 +3742,9 @@ sub rsync_backup_point {
         
         print_cmd(@cmd_stack);
         if (0 == $test) {
-            $result = system(@cmd_stack);
+            # silence gratuitous lvremove output
+            #$result = system(@cmd_stack);
+            $result = system(join " ", @cmd_stack, ">/dev/null");
             
             if ($result != 0) {
                 bail("Removal of LVM snapshot failed: $result");
